wrapper/sessions_test.go

```diff
@@ -0,0 +1,56 @@
+package wrapper
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	_ "github.com/joho/godotenv/autoload"
+)
+
+func TestDataRace(t *testing.T) {
+	bot := &Client{
+		// Set the Authentication Header using BotToken() or BearerToken().
+		Authentication: BotToken(os.Getenv("TOKEN")),
+		Authorization:  &Authorization{},
+		Config:         DefaultConfig(),
+		Handlers:       &Handlers{},
+	}
+
+	s := &Session{
+		mu: new(sessionMutex),
+	}
+
+	bot.Sessions = []*Session{s}
+	err := bot.Connect(bot.Sessions[0])
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	const heartbeats = 1
+	for {
+
+		if s.Connected == nil {
+			t.Fatalf("disconnected before calling disconnect")
+		}
+
+		if s.ID != "" {
```

Comment:
add reason for this if statement (to wait until we receive and handle the READY event).